### PR TITLE
Support custom `public/index.html` for esm-views

### PR DESCRIPTION
**Pull request by cristiano-belloni**
Allow custom index.html for esm-views. Includes a refactor of some loose function in a single step.
This enables index.html for esm-views built & started with Webpack + esm-views built & started with esbuild. Adding a `public/index.html` is optional and should be done by the user.

---

- [x] Write functional or snapshot tests for esm-views (currently if trampoline is not included, tests don't fail)
- [x] Packages of all type / build combination, excluding Webpack esm-views, build correctly
- [x] Webpack esm-views build correctly
- [x] Packages of all type / build combination, excluding Webpack esm-views, start correctly
- [x] Webpack esm-views start correctly
- [x] Write tests
- [x] Write jsdocs
- [x] Write docs

### CODE REVIEW

File: packages/modular-scripts/react-scripts/config/parts/esmViewConfig.js [hunk 1]
```diff
@@ -24,34 +27,59 @@ function createConfig({
   };
 }
 
-function createPluginConfig({ isEnvProduction, styleImports }) {
+function createPluginConfig({ isEnvProduction, styleImports, indexPath, env }) {
+  console.log(styleImports);
```
Comment (AlbertoBrusa) [inline packages/modular-scripts/react-scripts/config/parts/esmViewConfig.js:31]: Was this left intentionally?

**Review by AlbertoBrusa — APPROVED** [review]
looks good 

There will be conflicts with my PR but all will be easily fixed

File: docs/package-types/esm-view.md [hunk 2]
```diff
@@ -16,8 +16,6 @@ following essential differences:
 1. All their external `import`s are rewritten to point to an
    [ESM CDN](../esm-views/esm-cdn.md).
 
-1. They don't allow the user to specify a custom `index.html`
```
Comment (sgb-io) [inline docs/package-types/esm-view.md:19]: Do you also need to update the table on the main page? It currently has a cross emoji against `Custom index / assets`

File: packages/modular-scripts/react-scripts/config/parts/esmViewConfig.js [hunk 3]
```diff
@@ -24,34 +27,59 @@ function createConfig({
...
+    // Standard minimal template to start an esm-view
+    templateContent = dedent(`
+      <!DOCTYPE html>
+        <head>
+          ${styleImports?.map(
+            (importUrl) =>
+              `<link rel="stylesheet" href="${importUrl}"></script>`,
+          )}
+        </head>
+        <html>
```
Comment (sgb-io) [inline packages/modular-scripts/react-scripts/config/parts/esmViewConfig.js:63]: Do you have a duplicate `html` tag here? Looks like it

Comment (sgb-io) [reply packages/modular-scripts/react-scripts/config/parts/esmViewConfig.js:63]: Not new to your code but looks like an existing issue. Presumably the browser is automatically fixing this but it looks like invalid html to me (the html tag should wrap everything except the doctype)

File: packages/modular-scripts/react-scripts/config/parts/esmViewConfig.js (hunk 1 above)
Comment (cristiano-belloni) [reply packages/modular-scripts/react-scripts/config/parts/esmViewConfig.js]: No, thanks for catching this

File: packages/modular-scripts/react-scripts/config/parts/esmViewConfig.js (hunk 3 above)
Comment (cristiano-belloni) [reply packages/modular-scripts/react-scripts/config/parts/esmViewConfig.js:56]: Looks like a bug - moved

File: docs/package-types/esm-view.md (hunk 2 above)
Comment (cristiano-belloni) [reply docs/package-types/esm-view.md:19]: Good catch - done!

**Pull request merged**